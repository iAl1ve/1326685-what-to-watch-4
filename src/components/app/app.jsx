import React, {PureComponent} from "react";
import {Switch, Route, BrowserRouter} from "react-router-dom";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer.js";
import Main from "../main/main.jsx";
import MoviePage from "../movie-page/movie-page.jsx";
import {AppType} from '../../types/index.js';

class App extends PureComponent {
  _getCommentsById(id) {
    const {listReviews} = this.props;
    return listReviews.filter((review) => review.idFilm === id);
  }

  _renderAppScreen() {
    const {listMovies, currentGenre, activeFilm, listGenres, onTitleButtonClick, onGenreItemClick} = this.props;

    if (activeFilm === null) {
      return (
        <Main
          movie = {listMovies[0]}
          listMovies = {listMovies}
          onTitleButtonClick = {onTitleButtonClick}
          currentGenre = {currentGenre}
          listGenres = {listGenres}
          onGenreItemClick = {onGenreItemClick}
        />
      );
    }

    if (activeFilm) {
      return (
        <MoviePage
          movie = {activeFilm}
          listMovies = {listMovies}
          listReviews = {this._getCommentsById(activeFilm.id)}
          onTitleButtonClick = {onTitleButtonClick}
          currentGenre = {currentGenre}
        />
      );
    }

    return null;
  }

  render() {
    const {currentGenre, listMovies, onTitleButtonClick} = this.props;

    return (
      <BrowserRouter>
        <Switch>
          <Route exact path="/">
            {this._renderAppScreen()}
          </Route>
          <Route exact path="/dev-film">
            <MoviePage
              movie = {listMovies[0]}
              listMovies = {listMovies}
              listReviews = {this._getCommentsById(listMovies[0].id)}
              onTitleButtonClick = {onTitleButtonClick}
              currentGenre = {currentGenre}
            />
          </Route>
        </Switch>
      </BrowserRouter>
    );
  }
}

App.propTypes = AppType;

const mapStateToProps = (state) => ({
  currentGenre: state.currentGenre,
  activeFilm: state.activeFilm,
  listMovies: state.listMovies,
  listReviews: state.listReviews,
  listGenres: state.listGenres,
});

const mapDispatchToProps = (dispatch) => ({
  onTitleButtonClick(film) {
    dispatch(ActionCreator.setActiveMovie(film));
  },
  onGenreItemClick(genre) {
    dispatch(ActionCreator.changeCurrentGenre(genre));
  }
});

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);

