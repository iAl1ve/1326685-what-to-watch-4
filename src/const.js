export const UrlPage = {
  MAIN: `main`,
  MOVIE_PAGE: `movie-page`,
};

export const VIDEO_SIZE = 280;

export const PLAYBACK_DELAY = 1000;

export const MAX_SHOW_MORE_FILMS = 8;
export const MAX_COUNT_FILMS = 4;

export const GENRE_DEFAULT = `All genres`;

export const Tabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const MONTH_NAMES = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`,
];
