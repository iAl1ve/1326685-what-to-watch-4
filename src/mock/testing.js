export const ListMovies = [
  {
    id: 1,
    title: `The Grand Budapest Hotel`,
    genre: `Comedies`,
    year: 1996,
    src: `img/the-grand-budapest-hotel-poster.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.9,
    ratingCount: 256,
    movieDescription: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    movieDirector: `Wes Andreson`,
    movieStarring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 122,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B41D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 2,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Drama`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.1,
    ratingCount: 146,
    movieDescription: `Fantastic Beasts: The Crimes of Grindelwald movie description text. Fantastic Beasts: The Crimes of Grindelwald movie description text. Fantastic Beasts: The Crimes of Grindelwald movie description text. `,
    movieDirector: `Wes Andreson`,
    movieStarring: [`Bill Murray`, `Willem Dafoe and other`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 131,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B41D81`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 3,
    title: `Bohemian Rhapsody`,
    src: `img/bohemian-rhapsody.jpg`,
    genre: `Comedies`,
    year: 2015,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 4.1,
    ratingCount: 195,
    movieDescription: `Bohemian Rhapsody movie description text. Bohemian Rhapsody movie description text. Bohemian Rhapsody movie description text. `,
    movieDirector: `Bill Murray`,
    movieStarring: [`Edward Norton`, `Jude Law`, `Willem Dafoe`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 103,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B31D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 4,
    title: `Macbeth`,
    src: `img/macbeth.jpg`,
    genre: `Drama`,
    year: 2014,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.3,
    ratingCount: 119,
    movieDescription: `Macbeth movie description text. Macbeth movie description text. Macbeth movie description text. Macbeth movie description text. `,
    movieDirector: `Macbeth`,
    movieStarring: [`Macbeth`, `Willem Dafoe`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 212,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B40D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 5,
    title: `Aviator`,
    src: `img/aviator.jpg`,
    genre: `Documentary`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.2,
    ratingCount: 249,
    movieDescription: `Aviator movie description text. Aviator movie description text. Aviator movie description text. Aviator movie description text.`,
    movieDirector: `Aviator`,
    movieStarring: [`Bill Murray`, `Aviator`, `Willem Dafoe`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 165,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#A31D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 6,
    title: `Moonrise Kingdom`,
    src: `img/moonrise-kingdom.jpg`,
    genre: `Romance`,
    year: 2011,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.1,
    ratingCount: 232,
    movieDescription: `Moonrise Kingdom movie description text. Moonrise Kingdom movie description text. Moonrise Kingdom movie description text. Moonrise Kingdom movie description text.`,
    movieDirector: `Moonrise Kingdom`,
    movieStarring: [`Moonrise Kingdom`, `Jude Law`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 138,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#C51D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 7,
    title: `Midnight Special`,
    src: `img/midnight-special.jpg`,
    genre: `Crime`,
    year: 2016,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.4,
    ratingCount: 212,
    movieDescription: `Midnight Special movie description text. Midnight Special movie description text. Midnight Special movie description text. Midnight Special movie description text.`,
    movieDirector: `Midnight Special`,
    movieStarring: [`Midnight Special`, `Jude Law`, `Willem Dafoe`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 154,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#D31D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 8,
    title: `War of the Worlds`,
    src: `img/war-of-the-worlds.jpg`,
    genre: `Drama`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.9,
    ratingCount: 201,
    movieDescription: `War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. `,
    movieDirector: `War of the Worlds`,
    movieStarring: [`War of the Worlds`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 141,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B40A8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 9,
    title: `Shutter Island`,
    src: `img/shutter-island.jpg`,
    genre: `Horror`,
    year: 2017,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 7.9,
    ratingCount: 175,
    movieDescription: `Shutter Island movie description text. Shutter Island movie description text. Shutter Island movie description text. Shutter Island movie description text. `,
    movieDirector: `Shutter Island`,
    movieStarring: [`Bill Murray`, `Shutter Island`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 109,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#A61D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }
];

export const ListReviews = [
  {
    id: 1,
    idUser: 2,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    date: `2016-12-24`,
    rating: 7.6,
  }, {
    id: 2,
    idUser: 2,
    text: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    author: `Matthew Lickona`,
    date: `2016-12-20`,
    rating: 8.3,
  }, {
    id: 3,
    idUser: 2,
    text: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    author: `Bill Goodykoontz`,
    date: `2015-11-18`,
    rating: 7.3,
  },
  {
    id: 4,
    idUser: 3,
    text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    author: `Paula Fleri-Soler`,
    date: `2016-12-20`,
    rating: 7.8,
  },
  {
    id: 5,
    idUser: 1,
    text: `When one of Gustave's lovers dies mysteriously. It is certainly a magical and childlike way of storytelling.`,
    author: `Amanda Soler`,
    date: `2016-12-20`,
    rating: 9.6,
  },
  {
    id: 6,
    idUser: 3,
    text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    author: `Amanda Greever`,
    date: `2015-11-18`,
    rating: 8.9,
  },
  {
    id: 7,
    idUser: 4,
    text: `Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Bill Muir`,
    date: `2016-08-14`,
    rating: 8.4,
  },
  {
    id: 8,
    idUser: 5,
    text: `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs.`,
    author: `Wes  Dafoe`,
    date: `2015-11-21`,
    rating: 5.6,
  },
  {
    id: 9,
    idUser: 2,
    text: `Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    author: `Kate Norton`,
    date: `2015-03-08`,
    rating: 2,
  },
  {
    id: 10,
    idUser: 1,
    text: `Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    author: `Andreson Fleri`,
    date: `2016-05-09`,
    rating: 10,
  },
];

export const listGenres = [`All genres`, `Comedies`, `Drama`, `Documentary`, `Romance`, `Crime`, `Horror`];

export const notAdaptedListReviews = [
  {
    id: 1,
    user: {
      id: 2,
      name: `Kate Muir`
    },
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2016-12-24`,
    rating: 7.6,
  }
];

export const adaptedUserInfo = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};
