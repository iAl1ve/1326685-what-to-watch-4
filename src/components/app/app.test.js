import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {App} from "./app.jsx";
import NameSpace from "../../reducer/name-space.js";
import {ListMovies, listGenres, ListReviews} from "../../mock/testing.js";
import {AuthorizationStatus} from "../../const.js";
import {GENRE_DEFAULT, MAX_SHOW_MORE_FILMS} from "../../const.js";

const mockStore = configureStore([]);

describe(`Test Render App`, () => {
  it(`Render App`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        listMovies: ListMovies,
        promoFilm: ListMovies[0],
        listGenres,
        listReviews: ListReviews,
      },
      [NameSpace.APP_STATE]: {
        currentGenre: GENRE_DEFAULT,
        activeFilm: null,
        countShowMovies: MAX_SHOW_MORE_FILMS,
        isPlaying: false,
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
        userInfo: {},
        isStatusSend: false,
        isFormDisabled: false,
      }
    });

    const tree = renderer
      .create(
          <Provider store = {store}>
            <App
              movie = {ListMovies[0]}
              listMovies = {ListMovies}
              listGenres = {listGenres}
              listReviews = {ListReviews}
              activeFilm = {ListMovies[0]}
              currentGenre = {GENRE_DEFAULT}
              countShowMovies = {MAX_SHOW_MORE_FILMS}
              isAuthorization = {false}
              isFormDisabled = {false}
              onTitleButtonClick = {() => {}}
              onGenreItemClick = {() => {}}
              onShowMoreClick = {() => {}}
              onPlayerExitClick = {() => {}}
              onPlayButtonClick = {() => {}}
              login = {() => {}}
              onSubmitReview = {() => {}}
            />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
