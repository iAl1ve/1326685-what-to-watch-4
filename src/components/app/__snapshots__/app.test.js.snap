// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Render App Render App 1`] = `null`;
